server/src/main/java/com/google/musicanalysis/site/YoutubeServlet.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Sample Java code for youtube.videos.list
+ * See instructions for running these code samples locally:
+ * https://developers.google.com/explorer-help/guides/code_samples#java
+ */
+package com.google.musicanalysis.site;
+import com.google.musicanalysis.util.Secrets;
+import com.google.musicanalysis.util.URLEncodedBuilder;
+
+import com.google.gson.JsonParser;
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpResponse.BodyHandlers;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+import javax.servlet.annotation.WebServlet;
+
+
+@WebServlet("/api/youtube")
+public class YoutubeServlet extends HttpServlet {
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse res) 
+        throws ServletException, IOException {
+        // retrieve Youtube API key and access token 
+        String API_KEY = Secrets.getSecretString("YOUTUBE_API_KEY");
+        HttpSession session = req.getSession();
+        var accessToken = session.getAttribute("oauth-access-token-youtube");
+        if (session == null || accessToken == null) {
```

Comment:
You don't need to check if `session` is null here. It would have already thrown an error on line 33 if it was.